core/src/components/datetime/datetime.tsx

```diff
@@ -2001,6 +2029,37 @@ export class Datetime implements ComponentInterface {
               }
             }
 
+            let dateStyle: DatetimeHighlightStyle | undefined = undefined;
+
+            /**
+             * Custom highlight styles should not override the style for selected dates,
+             * nor apply to "filler days" at the start of the grid.
+             */
+            if (highlightedDates !== undefined && !isActive && day !== null) {
+              if (Array.isArray(highlightedDates)) {
+                const matchingHighlight = highlightedDates.find((hd) => hd.date === dateIsoString);
+                if (matchingHighlight) {
+                  dateStyle = {
+                    color: matchingHighlight.color,
+                    backgroundColor: matchingHighlight.backgroundColor,
+                  } as DatetimeHighlightStyle;
+                }
+              } else {
+                /**
+                 * Wrap in a try-catch to prevent exceptions in the user's function
+                 * from interrupting the calendar's rendering.
+                 */
+                try {
+                  dateStyle = highlightedDates(dateIsoString);
+                } catch (e) {
+                  printIonError(
+                    'Exception thrown from provided `highlightedDates` callback. Please check your function and try again.',
+                    e
```

Comment:
Can we additionally log the host element reference with this exception, so they can quickly find the `ion-datetime` instance this came from, if they have multiple rendered on the view?